lib/iris/fileformats/netcdf.py

```diff
@@ -1339,6 +1339,27 @@ def _get_dim_names(self, cube):
                 dimension_names.append(dim_name)
         return dimension_names
 
+    @staticmethod
+    def cf_valid_var_name(var_name):
+        """
+        Return a valid CF var_name given a potentially invalid name.
+
+        Args:
+
+        * var_name (str):
+            The var_name to normalise
+
+        Returns:
+            A var_name suitable for passing through for variable creation.
+
+        """
+        # Replace invalid charaters with an underscore ("_").
+        var_name = re.sub(r'[^a-zA-Z0-9]', "_", var_name)
```

Comment:
I think you could just use `r'[\W]'` here. And what about unicode characters?

Reply:
Personally, I can never rembember if ``\W`` contains the ``-`` character. Given that, my preference is explicit. Happy to be convinced otherwise though.